### PR TITLE
Chore: remove insertElementAtPath_DEPRECATED from reparent

### LINKED ISSUE #3637: Remove insertElementAtPath_DEPRECATED from reparent

### PR DESCRIPTION
**Pull request by ruggi**
Fixes #3637 

Removed `insertElementAtPath_DEPRECATED` from `getReparentOutcome` and `runReparentElement`.

### CODE REVIEW

**Review by Rheeseyb — APPROVED** [review]

**Review by bkrmendy — APPROVED** [review]

File: editor/src/components/canvas/commands/reparent-element-command.spec.tsx [hunk 1]
```diff
@@ -83,12 +78,7 @@ describe('runReparentElement', () => {
     ])
     const originalEditorState = renderResult.getEditorState().editor
 
-    const reparentCommand = reparentElement(
-      'always',
-      targetPath,
-      childInsertionPath(newParentPath),
-      'use-deprecated-insertJSXElementChild',
-    )
+    const reparentCommand = reparentElement('always', targetPath, childInsertionPath(newParentPath))
```
Comment (gbalint) [inline editor/src/components/canvas/commands/reparent-element-command.spec.tsx:81]: Is it sure that this is always a childInsertionPath?

Comment (bkrmendy) [reply editor/src/components/canvas/commands/reparent-element-command.spec.tsx:81]: no, because PASTE_JSX_ELEMENT has a big chunk of code that deals with wrapping elements into fragments on paste

**Review by bkrmendy — CHANGES_REQUESTED** [review]
Balint's comment (https://github.com/concrete-utopia/utopia/pull/3638#discussion_r1184986905) is a necessity rather than a nice to have, and on a related note, we need tests that cover pasting into an non-empty conditional branch. Without covering that case, merging this PR would regress wrapping into fragment on paste in conditionals, so that should be fixed before merge.

File: editor/src/components/canvas/commands/reparent-element-command.spec.tsx (hunk 1 above)
Comment (ruggi) [reply editor/src/components/canvas/commands/reparent-element-command.spec.tsx:81]: these more poignant tests should address that https://github.com/concrete-utopia/utopia/pull/3638/commits/31a96a0a3ecce576c5f9a5b5f732e8e01f282abb (cc @bkrmendy )
